modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ExtensionMethods.cs

```diff
@@ -26,6 +26,12 @@ public static bool IsGodotToolsProject(this GeneratorExecutionContext context)
                toggle != null &&
                toggle.Equals("true", StringComparison.OrdinalIgnoreCase);
 
+        public static bool IsGodotSourceGeneratorDisabled(this GeneratorExecutionContext context, string generatorName) =>
+            AreGodotSourceGeneratorsDisabled(context) ||
+            (context.TryGetGlobalAnalyzerProperty("GodotDisabledSourceGenerators", out string? disabledGenerators) &&
+            disabledGenerators != null &&
```

Comment:
Can the string be empty, or would it return null in that case? If it can be empty, we should use `IsNullOrEmpty`.

Reply:
I think it could be empty, but the split below would return an empty array in that case so it's not strictly necessary. 

I tried it, however `!string.IsNullOrEmpty(disabledGenerators)` strangely doesn't seem to pass the non-nullablility through to the following line so I'm inclined to leave it as is. It seems like there are other nullability warnings in this project already though so I'm happy to push it if you prefer @neikeq ?

Reply:
The `string.IsNullOrEmpty` method is not annotated for nullability in `netstandard2.0` which is the TFM that source generators must target.

Reply:
I believe checking if it's empty early should be faster than going through the `Split` and `Contains` calls, at the very least because those are not inlined.

I don't remember how we deal with this nullability issue elsewhere. We probably check for `null` and `Length` manually.

Reply:
What's the status here? I can't tell from this discussion if the PR should be merged or should be worked on further.

Reply:
This discussion is about saving one allocation and maybe 100 instructions executed in a code path that is executed exactly 7 times for each build of the csharp user project (saving like a microsecond on an action usually taking a few seconds) so it doesn't really matter all that much.

Reply:
I approved the PR, this requested change is optional.

> _(...)_ a code path that is executed exactly 7 times for each build of the csharp user project _(...)_ so it doesn't really matter all that much.

Source generators are also run very frequently by IDEs while editing code, so performance is very important. Still, just nitpicking in this case.